src/functional/iterator/transforms/eta_reduction.py

```diff
@@ -5,7 +5,11 @@
 class EtaReduction(NodeTranslator):
     """Eta reduction: simplifies `λ(args...) → f(args...)` to `f`."""
 
-    def visit_Lambda(self, node: ir.Lambda) -> ir.Node:
+    def visit_Lambda(self, node: ir.Lambda, **kwargs) -> ir.Node:
```

Comment:
```suggestion
    def visit_Lambda(self, node: ir.Lambda, *, in_reduce: bool) -> ir.Node:
```
Maybe let's implement it like this, because often we use kwargs to propagate info to any node, if someone at some point does the `self.generic_visit(node, **kwargs)` somewhere in this function, we will propagate `in_reduce` accidentally.

Reply:
Sorry, but how do I do this? Wouldn't this entail that `visit_Lambda` is only ever called "directly"? If it is called via the visitor mechanism I get

```
  File "/scratch-shared/meteoswiss/scratch/mroeth/icongt4py/gt4py/src/eve/visitors.py", line 123, in visit
    return visitor(node, **kwargs)
TypeError: EtaReduction.visit_Lambda() missing 1 required positional argument: 'in_reduce'
```

However, we are also required to be able to visit Lambdas that are not the body of a reduction...

Reply:
all non-explicit visits forward kwargs, would have to check exactly where it got lost.